internal/checker/checker.go

```diff
@@ -15087,7 +15087,7 @@ func (c *Checker) resolveEntityName(name *ast.Node, meaning ast.SymbolFlags, ign
 				name.Parent != nil && name.Parent.Kind == ast.KindJSExportAssignment) {
 			c.markSymbolOfAliasDeclarationIfTypeOnly(getAliasDeclarationFromName(name), symbol, nil /*finalTarget*/, true /*overwriteEmpty*/, nil, "")
 		}
-		if symbol.Flags&meaning == 0 && !dontResolveAlias {
+		if symbol.Flags&meaning == 0 && !dontResolveAlias && symbol.Flags&ast.SymbolFlagsAlias != 0 {
```

Comment:
In strada, `checkExpressionCached` on `a.b.c.d` caches resolved symbols on `a.b.c` and `a.b` and `a`. `getConstantValue` uses these cached symbols to shortcut its' logic. In corsa, it does not, and, in so doing, reveals that here we try to call `resolveAlias` on non-alias (module) symbols when we consider if we want to inline `module.exports`. This fixes that.